### ISSUE TITLE
Design accessible experience for going from filters to table

### ISSUE THREAD

**Issue by kimallen** [cisagov/manage.get.gov#5411]
### Issue description

_As a screen reader user, I need to know what filters I've applied to my domains table so that I can understand that I've filtered the table and what that filter has produced._

Our screenreader experience when filtering the domains table is not accessible. The current problems are:
- When closing the status filter on the domains page and tabbing back to the domain table a SR does not announce that the table is filtered in any way.
- To review what filters are applied, there is no summary, and an SR user needs to walk through the dropdown checkboxes to know what is checked.
- There is not (or may not be? needs testing) a way for the user to know if there are no results found

There should be some way to let the user know what is filtered in the table. 

### Acceptance criteria
Create documentation and guidance for the SR experience for filtered tables that represent the following scenarios
- [ ] A filter is applied to a table
- [ ] The filter produced no results 
- [ ] Create an implementation ticket for devs

### Additional context

Issue raised in PR #5374 in [this comment](https://github.com/cisagov/manage.get.gov/pull/5380#issuecomment-5670376517)

One common practice is to announce the results of the applied filter, like "x items found", and one option is that that summary could include the filters that are included.

**Comment by katypies** [comment]
Moving to ~dev~ design parking lot for sizing - I'd like to get a sense of how long this would take, so we can appropriately slot it into a coming sprint.
